pkg/apis/serving/v1alpha1/revision_validation.go

```diff
@@ -133,6 +135,17 @@ func validateContainer(container corev1.Container) *apis.FieldError {
 	return errs
 }
 
+func validateContainerPorts(ports []corev1.ContainerPort) bool {
+	bIsAllowed := false
+	// users can set container port which names "user-port" to define application's port.
+	// Queue-proxy will use it to send requests to application
+	if len(ports) == 1 && ports[0].Name == "user-port"{
```

Comment:
In https://github.com/knative/serving/issues/2258, @evankanderson didn't ask specifically what port name is required for `port[0]` but it is probably ok to require it here.

However, can we please make the validation message clearly indicate the failures, i.e.:
* In case user specifies more than one port, complain that only one port is allowed.
* In case user specifies a port with a disallowed name, complain that the name has to be "user-port".  Or alternatively we can drop this requirement.

Also, if we want to keep the port name requirement, can  we please move this constant from pkg/reconciler/v1alpha1/revision/resources/constants.go may be to revision_type.go?

Thanks

Reply:
Yes, you are right, it will more friendly to user. I will fix it. Thank you.